### PR TITLE
backingchain:add case for domblkthreshold with backing chain element

### PR DESCRIPTION
**Pull request by nanli1**
VIRT-294599: Do domblkthreshold for device with backing chain element
Signed-off-by: nanli <nanli@redhat.com>

Test result: 
```
/usr/local/bin/avocado run --vt-type libvirt --test-runner=runner --vt-machine-type q35  backingchain.virsh_domblk
 (1/2) type_specific.io-github-autotest-libvirt.backingchain.virsh_domblk_test.backing_target: PASS (67.25 s)
 (2/2) type_specific.io-github-autotest-libvirt.backingchain.virsh_domblk_test.entire_disk: PASS (75.23 s)
```

### CODE REVIEW

File: provider/backingchain/blockcommand_base.py [hunk 1]
```diff
@@ -258,10 +258,14 @@ def prepare_secret_disk(self, image_path, secret_disk_dict=None):
...
+    def clean_file(file_path, session=None):
         """
         Clean file
+
         :params file_path: the path of file to delete
+        :params session: vm session
         """
+        if session:
+            session.cmd("rm -f %s" % file_path)
         if os.path.exists(file_path):
```
Comment (Yingshun) [inline provider/backingchain/blockcommand_base.py:270]: should you update the 2nd 'if' to 'elif'? 
```suggestion
        if session:
            session.cmd("rm -f %s" % file_path)
        elif os.path.exists(file_path):
```

File: libvirt/tests/src/backingchain/virsh_domblk_test/domblkthreshold_with_backingchain_element.py [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+        result = virsh.domstats(vm_name, options, debug=True,
+                                ignore_status=False).stdout.strip()
+
+        pattern = r"block.*.threshold=%s" % threshold_value
+        if not threshold_value:
+            if re.search(pattern, result):
+                test.fail("Threshold: %s should not be in %s" % (pattern, result))
+        else:
+            if not re.search(pattern, result):
+                test.fail("Not get correct threshold: %s be in %s" % (pattern, result))
```
Comment (Yingshun) [inline libvirt/tests/src/backingchain/virsh_domblk_test/domblkthreshold_with_backingchain_element.py:88]:
```suggestion
                test.fail("Not get correct threshold: %s should be in %s" % (pattern, result))
```

File: libvirt/tests/src/backingchain/virsh_domblk_test/domblkthreshold_with_backingchain_element.py [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+
+    def check_domstats_threshold(options, threshold_value=None):
+        """
+        Check domstats threshold
+
+        :param options: extra options for virsh domstats command
+        :param threshold_value: domstats threshold value
+        """
+        result = virsh.domstats(vm_name, options, debug=True,
+                                ignore_status=False).stdout.strip()
```
Comment (Yingshun) [inline libvirt/tests/src/backingchain/virsh_domblk_test/domblkthreshold_with_backingchain_element.py:80]:
```suggestion
        result = virsh.domstats(vm_name, options, debug=True,
                                ignore_status=False).stdout_text.strip()
```

File: libvirt/tests/src/backingchain/virsh_domblk_test/domblkthreshold_with_backingchain_element.py [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+
+        pattern = r"block.*.threshold=%s" % threshold_value
```
Comment (Yingshun) [inline libvirt/tests/src/backingchain/virsh_domblk_test/domblkthreshold_with_backingchain_element.py:82]: It seems that this line should be under L83. For default value(None), is it correct to check if there is no 'block.*.threshold=None'??

File: provider/backingchain/blockcommand_base.py (hunk 1 above)
Comment (nanli1) [reply provider/backingchain/blockcommand_base.py:270]: 还真是

Comment (nanli1) [reply provider/backingchain/blockcommand_base.py]: Thanks your patience ,yingshun Done :D,

File: libvirt/tests/src/backingchain/virsh_domblk_test/domblkthreshold_with_backingchain_element.py (hunk 2 above)
Comment (nanli1) [reply libvirt/tests/src/backingchain/virsh_domblk_test/domblkthreshold_with_backingchain_element.py]: done

File: libvirt/tests/src/backingchain/virsh_domblk_test/domblkthreshold_with_backingchain_element.py (hunk 3 above)
Comment (nanli1) [reply libvirt/tests/src/backingchain/virsh_domblk_test/domblkthreshold_with_backingchain_element.py]: done

File: libvirt/tests/src/backingchain/virsh_domblk_test/domblkthreshold_with_backingchain_element.py (hunk 4 above)
Comment (nanli1) [reply libvirt/tests/src/backingchain/virsh_domblk_test/domblkthreshold_with_backingchain_element.py]: OK ,Updated :D

**Review by Yingshun — APPROVED** [review]

File: libvirt/tests/src/backingchain/virsh_domblk/domblkthreshold_with_backingchain_element.py [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+        session.close()
+
+    def check_event(event_session, expected_event):
+        """
+        Check event correct
+
+        :param event_session: virsh session
+        :param expected_event: expected event pattern
+        """
+        test.log.debug('Checking event pattern is :%s ' % expected_event)
```
Comment (dzhengfy) [inline libvirt/tests/src/backingchain/virsh_domblk/domblkthreshold_with_backingchain_element.py:107]: %  -> ,

**Review by dzhengfy — COMMENTED** [review]
Other LGTM

File: libvirt/tests/src/backingchain/virsh_domblk/domblkthreshold_with_backingchain_element.py (hunk 5 above)
Comment (nanli1) [reply libvirt/tests/src/backingchain/virsh_domblk/domblkthreshold_with_backingchain_element.py:107]: Sorry , I didn't get it . Do you mean     `  test.log.debug('Checking event pattern is   -> %s  ' % expected_event)` @dzhengfy

Comment (dzhengfy) [reply libvirt/tests/src/backingchain/virsh_domblk/domblkthreshold_with_backingchain_element.py:107]: yes.

**Review by dzhengfy — APPROVED** [review]

**Pull request merged**